system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/RegistrationApiClientTest.java

```diff
@@ -38,62 +42,88 @@
 import java.util.Map;
 import java.util.UUID;
 
+import static java.time.temporal.ChronoUnit.SECONDS;
 import static java.util.concurrent.TimeUnit.MINUTES;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.awaitility.Awaitility.await;
-import static org.eclipse.dataspaceconnector.registration.client.RegistrationServiceTestUtils.CLIENT_DID_WEB;
-import static org.eclipse.dataspaceconnector.registration.client.RegistrationServiceTestUtils.UNREGISTERED_CLIENT_DID_WEB;
+import static org.eclipse.dataspaceconnector.junit.testfixtures.TestUtils.getFreePort;
+import static org.eclipse.dataspaceconnector.registration.client.RegistrationServiceTestUtils.didDocument;
+import static org.mockserver.integration.ClientAndServer.startClientAndServer;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static org.mockserver.stop.Stop.stopQuietly;
 
 @IntegrationTest
 public class RegistrationApiClientTest {
     static final String API_URL = "http://localhost:8182/authority";
-    static final Faker FAKER = new Faker();
     static final Monitor MONITOR = new ConsoleMonitor();
     public static final String HUB_BASE_URL = "http://localhost:8181/api/identity-hub";
-    static RegistryApi api;
+
     static IdentityHubClientImpl identityHubClient;
 
-    String participantUrl = FAKER.internet().url();
-    Instant startTime = Instant.now();
+    int apiPort;
+    ClientAndServer httpSourceClientAndServer;
 
     @BeforeAll
     static void setUpClass() {
-        var apiClient = ClientUtils.createApiClient(API_URL, CLIENT_DID_WEB, TestKeyData.PRIVATE_KEY_P256);
-        api = new RegistryApi(apiClient);
         var okHttpClient = new OkHttpClient.Builder().build();
         identityHubClient = new IdentityHubClientImpl(okHttpClient, new ObjectMapper(), MONITOR);
     }
 
+    @BeforeEach
+    void setUp() throws Exception {
+        apiPort = getFreePort();
+        httpSourceClientAndServer = startClientAndServer(apiPort);
+        httpSourceClientAndServer.when(request().withPath("/.well-known/did.json"))
+                .respond(response()
+                        .withBody(didDocument())
+                        .withStatusCode(HttpStatusCode.OK_200.code()));
+    }
+
+    @AfterEach
+    void tearDown() {
+        stopQuietly(httpSourceClientAndServer);
+    }
+
     @Test
     void listParticipants() {
+        var did = did();
+        var api = api(did);
```

Comment:
any reason we can't instantiate these two variables in the `@BeforeEach`? seems less repetitive

Reply:
good point, moved to @BeforeEach